### PR TITLE
feat(PauseableImage): :sparkles: Creating new pausable image component

### PR DESCRIPTION
**Pull request by vonbarown**
### Overview

Creating a new image component that allows users to pause gifs to improve accessibility of gifs.

with monolith integration branch [PR](https://github.com/codecademy-engineering/Codecademy/pull/29727)

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [GM-333](https://codecademy.atlassian.net/browse/GM-333)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.196967.0<br/>@codecademy/gamut-labs@29.3.6-alpha.196967.0<br/>@codecademy/styleguide@54.3.6-alpha.196967.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.6bbe07.0<br/>@codecademy/gamut-labs@29.3.6-alpha.6bbe07.0<br/>@codecademy/styleguide@54.3.6-alpha.6bbe07.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6218f77c9a5fce0be28ed9a1--gamut-preview.netlify.app>https://6218f77c9a5fce0be28ed9a1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6218f77c9a5fce0be28ed9a1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.fe5575.0<br/>@codecademy/gamut-labs@29.3.6-alpha.fe5575.0<br/>@codecademy/styleguide@54.3.6-alpha.fe5575.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6218fc6dcaeed6155ee02981--gamut-preview.netlify.app>https://6218fc6dcaeed6155ee02981--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6218fc6dcaeed6155ee02981>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.ca9385.0<br/>@codecademy/gamut-labs@29.3.6-alpha.ca9385.0<br/>@codecademy/styleguide@54.3.6-alpha.ca9385.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.828773.0<br/>@codecademy/gamut-labs@29.3.6-alpha.828773.0<br/>@codecademy/styleguide@54.3.6-alpha.828773.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62190290a4341a1c60de5244--gamut-preview.netlify.app>https://62190290a4341a1c60de5244--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62190290a4341a1c60de5244>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.6e985d.0<br/>@codecademy/gamut-labs@29.3.6-alpha.6e985d.0<br/>@codecademy/styleguide@54.3.6-alpha.6e985d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62190f61a4341a2d33de5497--gamut-preview.netlify.app>https://62190f61a4341a2d33de5497--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62190f61a4341a2d33de5497>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.5c2252.0<br/>@codecademy/gamut-labs@29.3.6-alpha.5c2252.0<br/>@codecademy/styleguide@54.3.6-alpha.5c2252.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://621912bf09e05f2dfb5ac962--gamut-preview.netlify.app>https://621912bf09e05f2dfb5ac962--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/621912bf09e05f2dfb5ac962>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.4923af.0<br/>@codecademy/gamut-labs@29.3.6-alpha.4923af.0<br/>@codecademy/styleguide@54.3.6-alpha.4923af.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62193737e58c5d19c066846f--gamut-preview.netlify.app>https://62193737e58c5d19c066846f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62193737e58c5d19c066846f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.92d1df.0<br/>@codecademy/gamut-labs@29.3.6-alpha.92d1df.0<br/>@codecademy/styleguide@54.3.6-alpha.92d1df.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://621949946dd14325f9c08e66--gamut-preview.netlify.app>https://621949946dd14325f9c08e66--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/621949946dd14325f9c08e66>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.65c5a7.0<br/>@codecademy/gamut-labs@29.3.6-alpha.65c5a7.0<br/>@codecademy/styleguide@54.3.6-alpha.65c5a7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62194d3f6dd1432a8fc08cda--gamut-preview.netlify.app>https://62194d3f6dd1432a8fc08cda--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62194d3f6dd1432a8fc08cda>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.d462db.0<br/>@codecademy/gamut-labs@29.3.6-alpha.d462db.0<br/>@codecademy/styleguide@54.3.6-alpha.d462db.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62194f47717f30338757ca4f--gamut-preview.netlify.app>https://62194f47717f30338757ca4f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62194f47717f30338757ca4f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.65-alpha.84947b.0<br/>@codecademy/gamut-labs@29.3.6-alpha.84947b.0<br/>@codecademy/styleguide@54.3.6-alpha.84947b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://621961d919798c498087d948--gamut-preview.netlify.app>https://621961d919798c498087d948--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/621961d919798c498087d948>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.67-alpha.7fe330.0<br/>@codecademy/gamut-labs@29.3.8-alpha.7fe330.0<br/>@codecademy/styleguide@54.3.8-alpha.7fe330.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://621d3f363ea8a53922308b1a--gamut-preview.netlify.app>https://621d3f363ea8a53922308b1a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/621d3f363ea8a53922308b1a>Deploy Logs</a></p>

File: packages/gamut-labs/typings/react-freezeframe.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+declare module 'react-freezeframe' {
+  export interface FreezeframeProps {
+    className?: string;
+    alt?: string;
+    src: string;
+  }
+
+  export default class Freezframe extends React.Component<FreezeframeProps> {}
```
Comment (dreamwasp) [inline packages/gamut-labs/typings/react-freezeframe.d.ts:8]: type: Freezeframe

File: packages/gamut-labs/src/PausableImage/BaseImage/styles.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+
+export const imageStyles = css`
+  height: auto;
+  max-height: 100%;
+  max-width: 100%;
+`;
```
Comment (dreamwasp) [inline packages/gamut-labs/src/PausableImage/BaseImage/styles.ts:8]: i think ideally some of these would use [variants methods](https://gamut.codecademy.com/?path=/docs/foundations-system-variants--page), but honestly i'm not familiar enough with react freezeframe to know if that's even possible

File: packages/gamut-labs/src/PausableImage/BaseImage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    ? ['Play animated image', PlayIcon, StyledFreezeframe]
+    : ['Pause animated image', PauseIcon, PlayingImage];
+
+  return (
+    <Container>
+      <Image {...props} />
+      <FillButton
+        bottom={0}
+        m={8}
+        mode="light"
```
Comment (dreamwasp) [inline packages/gamut-labs/src/PausableImage/BaseImage/index.tsx:28]: i don't think you need the mode here

File: packages/gamut-labs/src/PausableImage/BaseImage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        onClick={() => setPaused(!paused)}
+        position="absolute"
+        right={0}
+        variant="secondary"
+        zIndex={1}
+        aria-label={buttonLabel}
+      >
+        <Icon color="white" />
```
Comment (dreamwasp) [inline packages/gamut-labs/src/PausableImage/BaseImage/index.tsx:36]: we probably want this set to `text` so it can be responsive to its ColorMode

**Review by dreamwasp — COMMENTED** [review]
looks good! will test in portal + monolith once the mode stuff is swapped over

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.67-alpha.88c6ef.0<br/>@codecademy/gamut-labs@29.3.8-alpha.88c6ef.0<br/>@codecademy/styleguide@54.3.8-alpha.88c6ef.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://621fd6b302034b1c5d244fd5--gamut-preview.netlify.app>https://621fd6b302034b1c5d244fd5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/621fd6b302034b1c5d244fd5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.69-alpha.367e3d.0<br/>@codecademy/gamut-labs@30.0.1-alpha.367e3d.0<br/>@codecademy/styleguide@55.0.1-alpha.367e3d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6220e0c5f22b070679164e8d--gamut-preview.netlify.app>https://6220e0c5f22b070679164e8d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6220e0c5f22b070679164e8d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.69-alpha.0154bc.0<br/>@codecademy/gamut-labs@30.0.1-alpha.0154bc.0<br/>@codecademy/styleguide@55.0.1-alpha.0154bc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6220e7b3e515300947ceb400--gamut-preview.netlify.app>https://6220e7b3e515300947ceb400--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6220e7b3e515300947ceb400>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6220eb28e51530084aceb4f2--gamut-preview.netlify.app>https://6220eb28e51530084aceb4f2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6220eb28e51530084aceb4f2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.69-alpha.d85c80.0<br/>@codecademy/gamut-labs@30.0.1-alpha.d85c80.0<br/>@codecademy/styleguide@55.0.1-alpha.d85c80.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6221139be515303a54ceb467--gamut-preview.netlify.app>https://6221139be515303a54ceb467--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6221139be515303a54ceb467>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
sweet, looks great!

File: packages/styleguide/stories/Molecules/PausableImage.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  <Story name="Dark mode" height="500px" args={{ mode: 'dark' }}>
+    {(args) => (
+      <ColorMode {...args}>
+        <PausableImage
+          src="//images.ctfassets.net/go6kr6r0ykrq/2vjTCK9DmJ6BOd5NtFxNIG/e5ced4209aef5241cd581841fa6e2511/learn-css_box-model_06.16.gif"
+          alt=""
+        />
+      </ColorMode>
+    )}
+  </Story>
```
Comment (dreamwasp) [inline packages/styleguide/stories/Molecules/PausableImage.stories.mdx:74]: you don't need to add these examples, i'd just refer folks to the colormode changer on the top of the page

**Review by dreamwasp — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.69-alpha.cffec1.0<br/>@codecademy/gamut-labs@30.0.1-alpha.cffec1.0<br/>@codecademy/styleguide@55.0.1-alpha.cffec1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6227e1cd92726c18f871cac6--gamut-preview.netlify.app>https://6227e1cd92726c18f871cac6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6227e1cd92726c18f871cac6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.74-alpha.7091e4.0<br/>@codecademy/gamut-labs@30.1.3-alpha.7091e4.0<br/>@codecademy/styleguide@55.0.6-alpha.7091e4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6228c20b44de5c0ab6a42a88--gamut-preview.netlify.app>https://6228c20b44de5c0ab6a42a88--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6228c20b44de5c0ab6a42a88>Deploy Logs</a></p>

**Pull request merged**